src/systems/filecoin_blockchain/vm/message/message.id

```diff
@@ -14,30 +14,30 @@ type Message union {
 }  // representation keyed
 
 type UnsignedMessage struct {
-    to        addr.Address
-    from      addr.Address
+    From        addr.Address
+    To          addr.Address
+
+    Method      actor.MethodNum
+    Params      actor.MethodParams  // Serialized parameters to the method.
 
     // When receiving a message from a user account the nonce in the message must match
     // the expected nonce in the "from" actor. This prevents replay attacks.
-    nonce     actor.CallSeqNum
-    value     actor.TokenAmount
-
-    gasPrice  actor.TokenAmount
-    gasLimit  actor.TokenAmount
+    CallSeqNum  actor.CallSeqNum
+    Value       actor.TokenAmount
 
-    method    actor.MethodNum
-    params    actor.MethodParams  // Serialized parameters to the method.
+    GasPrice
```

Comment:
does this have no type?

Reply:
It does -- `GasPrice GasPrice`. the codeGen tool removes redundancy between name and type.